In [1]:
import pandas as pd
data = pd.read_csv(R'C:\Users\user\Desktop\HAMOYE\FoodBalanceSheets_E_Africa_NOFLAG.csv' )
data

Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     1000 persons  38924.00   
1                    Domestic supply quantity      1000 tonnes      0.00   
2               Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3      Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4          Fat supply quantity (g/capita/day)     g/capita/day     80.06   
...                                       ...              ...       ...   
60938                                    Food      1000 tonnes     42.00   
60939     Food supply quantity (kg/capita/yr)               kg      3.06   
60940           Food supply (kcal/capita/day)  kcal/capita/day      3.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day      0.10   
60942      Fat supply quantity (g/capita/day)     g/capita/day      0.04   

          Y2015     Y2016     Y2017     Y2018  
0      39728.00  40551.00  41389.00  42228.00  
1          0.00      0.00      0.00      0.00  
2       3379.00   3372.00   3341.00   3322.00  
3         94.35     94.72     92.82     91.83  
4         79.36     77.40     80.19     77.28  
...         ...       ...       ...       ...  
60938     46.00     33.00     19.00     16.00  
60939      3.33      2.35      1.33      1.08  
60940      4.00      3.00      1.00      1.00  
60941      0.11      0.08      0.04      0.04  
60942      0.05      0.03      0.02      0.01  

[60943 rows x 12 columns]

In [2]:
data.shape

(60943, 12)

In [3]:
data.isna().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

The years have different missing values

In [4]:
data.describe(include ='all')

Area Code   Area     Item Code                     Item  \
count   60943.000000  60943  60943.000000                    60943   
unique           NaN     49           NaN                      119   
top              NaN  Kenya           NaN  Milk - Excluding Butter   
freq             NaN   1560           NaN                     1262   
mean      134.265576    NaN   2687.176706                      NaN   
std        72.605709    NaN    146.055739                      NaN   
min         4.000000    NaN   2501.000000                      NaN   
25%        74.000000    NaN   2562.000000                      NaN   
50%       136.000000    NaN   2630.000000                      NaN   
75%       195.000000    NaN   2775.000000                      NaN   
max       276.000000    NaN   2961.000000                      NaN   

        Element Code                   Element         Unit          Y2014  \
count   60943.000000                     60943        60943   59354.000000   
unique           NaN                        18            5            NaN   
top              NaN  Domestic supply quantity  1000 tonnes            NaN   
freq             NaN                      5295        40933            NaN   
mean     3814.856456                       NaN          NaN     134.196282   
std      2212.007033                       NaN          NaN    1567.663696   
min       511.000000                       NaN          NaN   -1796.000000   
25%       684.000000                       NaN          NaN       0.000000   
50%      5142.000000                       NaN          NaN       0.090000   
75%      5511.000000                       NaN          NaN       8.340000   
max      5911.000000                       NaN          NaN  176405.000000   

                Y2015          Y2016          Y2017          Y2018  
count    59395.000000   59408.000000   59437.000000   59507.000000  
unique            NaN            NaN            NaN            NaN  
top               NaN            NaN            NaN            NaN  
freq              NaN            NaN            NaN            NaN  
mean       135.235966     136.555222     140.917765     143.758381  
std       1603.403984    1640.007194    1671.862359    1710.782658  
min      -3161.000000   -3225.000000   -1582.000000   -3396.000000  
25%          0.000000       0.000000       0.000000       0.000000  
50%          0.080000       0.080000       0.100000       0.070000  
75%          8.460000       8.430000       9.000000       9.000000  
max     181137.000000  185960.000000  190873.000000  195875.000000

## 1. What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

In [5]:
data.groupby('Item')['Y2014'].sum()

Item
Alcohol, Non-Food        2403.00
Alcoholic Beverages    102410.11
Animal Products         11935.65
Animal fats            209460.54
Apples and products      9499.23
                         ...    
Vegetables, Other      155038.96
Vegetal Products       107145.19
Wheat and products     232670.13
Wine                     4497.36
Yams                   200396.96
Name: Y2014, Length: 119, dtype: float64

### Animal Fat produced in Y2014 = 209460.54

In [6]:
data.groupby('Item')['Y2017'].sum()

Item
Alcohol, Non-Food        2348.00
Alcoholic Beverages     95581.06
Animal Products         11547.65
Animal fats            269617.53
Apples and products     10198.90
                         ...    
Vegetables, Other      157752.59
Vegetal Products       107655.20
Wheat and products     240047.62
Wine                     4178.02
Yams                   229174.59
Name: Y2017, Length: 119, dtype: float64

#### Animal Fat produced in Y2017 = 269617.53

##  Animal Fat produced in 2014 and 2017 are 209460.54 and 269617.53 respectively

## 2. What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [7]:
print(round(data.Y2015.mean(),3))
print(round(data.Y2015.std(),3))

135.236
1603.404


## The Mean and Standard deviation across the whole dataset for the year 2015 are 135.236 and 1603.404

## 3. What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [8]:
data.Y2016.isna().sum()

1535

In [9]:
round(((data.Y2016.isna().sum())/data.Y2016.count()) * 100, 2)

2.58

## The total number and percentage of missing data in 2016 to 2 decimal places are 1535 and 2.53 respectively

## 4. Which year had the highest correlation with ‘Element Code’? 

In [10]:
data.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

## From the correlation chart above year 2014 has the highest correlation with the Element code with correlation value = 0.024457

## 5. What year has the highest sum of Import Quantity?

In [11]:
A = data[data.Element == 'Import Quantity']
A.Y2014.sum(),A.Y2015.sum(), A.Y2016.sum(),A.Y2017.sum(),A.Y2018.sum()

(274144.48, 267018.45999999996, 286582.78, 294559.09, 287997.08999999997)

## The year 2017 has the highest sum of Import Quantity

## 6. What is the total number of the sum of Production in 2014?

In [12]:
data.groupby('Element')['Y2014'].sum()

Element
Domestic supply quantity                  1996716.35
Export Quantity                            150020.64
Fat supply quantity (g/capita/day)          10225.56
Feed                                       216927.89
Food                                      1212332.49
Food supply (kcal/capita/day)              454257.00
Food supply quantity (kg/capita/yr)         49650.63
Import Quantity                            274144.48
Losses                                     153223.00
Other uses (non-food)                       78718.13
Processing                                 282923.00
Production                                1931287.75
Protein supply quantity (g/capita/day)      11836.46
Residuals                                   30149.00
Seed                                        21922.92
Stock Variation                             58749.83
Total Population - Both sexes             1031585.00
Tourist consumption                           416.00
Name: Y2014, dtype: float64

## The total number of the sum of Production in 2014 is 1931287.75

## 7. Which of these elements had the highest sum in 2018?

In [13]:
data.groupby('Element')['Y2018'].sum()

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

## The Domestic supply quantity has the highest sum among the elements in 2018

## 8. Which of these elements had the 3rd lowest sum in 2018?

Protein supply quantity (g/capita/day)

Food supply (kcal/capita/day)

Fat supply quantity (g/capita/day)

Seed

Food supply quantity (kg/capita/yr)

In [14]:
data.groupby('Element')['Y2018'].sum()

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

## Among the listed options, Food supply (kcal/capita/day) has the third lowest quantity 

## 9. What is the total Import Quantity in Algeria in 2018?

In [15]:
# note A has been defined for import quantity before
A = data[data.Element == 'Import Quantity']
A.groupby("Area")['Y2018'].sum()

Area
Algeria                        36238.29
Angola                          7769.67
Benin                           7008.24
Botswana                        1908.42
Burkina Faso                    2605.64
Cabo Verde                       499.66
Cameroon                        4188.88
Central African Republic          75.02
Chad                             491.95
Comoros                            2.92
Congo                           1785.27
C�te d'Ivoire                   9271.21
Djibouti                        3784.92
Egypt                          53954.20
Eswatini                         936.53
Ethiopia                        5655.24
Ethiopia PDR                       0.00
Gabon                           1079.75
Gambia                          1212.10
Ghana                           9914.68
Guinea                          4707.95
Guinea-Bissau                    497.72
Kenya                          11188.96
Lesotho                         1016.97
Liberia                         120

##  The total Import Quantity in Algeria in 2018 is 36238.29

## 10. What is the total number of unique countries in the dataset?

In [16]:
len(set(data.Area))

49

## The total number of unique countries in the dataset is 49.